codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-endpoint-discovery-async.java

```diff
@@ -205,9 +239,19 @@ public final String serviceName() {
                             .withOperationName("TestDiscoveryOptional")
                             .withMarshaller(new TestDiscoveryOptionalRequestMarshaller(protocolFactory))
                             .withResponseHandler(responseHandler).withErrorResponseHandler(errorResponseHandler)
-                            .discoveredEndpoint(cachedEndpoint).withInput(testDiscoveryOptionalRequest));
+                            .withMetricCollector(apiCallMetricCollector).discoveredEndpoint(cachedEndpoint)
+                            .withInput(testDiscoveryOptionalRequest));
+            AwsRequestOverrideConfiguration requestOverrideConfig = testDiscoveryOptionalRequest.overrideConfiguration().orElse(
+                    null);
+            executeFuture.whenComplete((r, e) -> {
+                List<MetricPublisher> metricPublishers = resolveMetricPublishers(clientConfiguration, requestOverrideConfig);
+                metricPublishers.forEach(p -> p.publish(apiCallMetricCollector.collect()));
```

Comment:
Have we documented that throwing an exception from one publisher will abort the other publishers?